evm_arithmetization/src/poseidon/poseidon_stark.rs

```diff
@@ -28,20 +30,157 @@ use super::columns::{
 use crate::all_stark::{EvmStarkFrame, Table};
 use crate::witness::memory::MemoryAddress;
 
-pub(crate) fn ctl_looked<F: Field>() -> TableWithColumns<F> {
+/// Maximum number of bytes that can be packed into a field element without
+/// performing a modular reduction.
+// TODO: this constant depends on the size of F, which is not bounded.
+pub const FELT_MAX_BYTES: usize = 7;
+
+pub fn ctl_looked_data<F: Field>() -> Vec<Column<F>> {
+    let cols = POSEIDON_COL_MAP;
+    let outputs: Vec<Column<F>> = Column::singles(cols.digest).collect();
+    let mut res: Vec<_> = Column::singles([
+        cols.context,
+        cols.segment,
+        cols.virt,
+        cols.len,
+        cols.timestamp,
+    ])
+    .collect();
+    res.extend(outputs);
+    res
+}
+
+pub(crate) fn ctl_looked_simple_op<F: Field>() -> TableWithColumns<F> {
     let mut columns = Column::singles(POSEIDON_COL_MAP.input).collect_vec();
     columns.extend(Column::singles(POSEIDON_COL_MAP.digest));
+    columns.push(Column::single(POSEIDON_COL_MAP.is_simple_op));
     TableWithColumns::new(
         *Table::Poseidon,
         columns,
         Some(Filter::new_simple(Column::single(
-            POSEIDON_COL_MAP.not_padding,
+            POSEIDON_COL_MAP.is_simple_op,
         ))),
     )
 }
 
+pub(crate) fn ctl_looked_general_output<F: Field>() -> TableWithColumns<F> {
```

Comment:
I think you need to include the timestamp, to make sure to bind the output with the correct input.